### PR TITLE
Financial Transaction Apply First In First out extension

### PR DESCRIPTION
**Pull request by joe-explr**
Description
----
Adding Apply First in First Out extension to 'Basic' resource.

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

image
image

### CODE REVIEW

**Review by SantanuBanerjee — APPROVED** [review]

File: content/millennium/r4/financial/general/financial-transaction.md [hunk 1]
```diff
@@ -44,6 +45,7 @@ All URLs for custom extensions are defined as `https://fhir-ehr.cerner.com/r4/St
  `financial-transaction-method`          | [`string`]                        | Describes the method of payment for the financial transaction.
  `financial-transaction-tendered-amount` | [`Money`]                         | The amount of cash originally tendered for payment. This value should be greater than or equal to the amount of the cash payment.
  `financial-transaction-type`            | [`CodeableConcept`]               | The classification of the transaction.
+ `apply-first-in-first-out`              | [`boolean`]                       | Whether to apply FIFO logic for recurring encounters or not.
```
Comment (alexbeckner1) [inline content/millennium/r4/financial/general/financial-transaction.md:48]: I believe these should be in alphabetical order.

File: lib/resources/r4/financial_transaction.yaml [hunk 2]
```diff
@@ -291,3 +291,20 @@ fields:
...
+  description: Whether to apply FIFO logic for recurring encounters or not.
+  action: create
+  example: |
+    {
+      "url": "https://fhir-ehr.cerner.com/r4/StructureDefinition/apply-first-in-first-out",
+      "valueBoolean": true
+    }
+  note: >
+    <ul>
+      <li>The default value is boolean value 'true'.</li>
```
Comment (alexbeckner1) [inline lib/resources/r4/financial_transaction.yaml:307]: I think we need notes around how the payment will be applied in CPA when set to true or false:
When the value is 'true' the payment will be posted across existing self-pay balances.
When the value is 'false' the system creates a new recurring encounter and posts the payment to the new self pay balance.

File: content/millennium/r4/financial/general/financial-transaction.md [hunk 3]
```diff
@@ -27,6 +27,7 @@ The FinancialTransaction resource provides the ability to record and exchange fi
 * [Financial Transaction Method]
 * [Financial Transaction Tendered Amount]
 * [Financial Transaction Type]
+* [Apply First In First Out]
```
[the remark this answers is not in the document]
Comment (joe-explr) [reply content/millennium/r4/financial/general/financial-transaction.md]: Addressed in 10f7519981f3a6a5b0220a264d546cce6fc7fed5

File: content/millennium/r4/financial/general/financial-transaction.md (hunk 1 above)
Comment (joe-explr) [reply content/millennium/r4/financial/general/financial-transaction.md]: Addressed in 10f7519981f3a6a5b0220a264d546cce6fc7fed5

File: lib/resources/r4/financial_transaction.yaml (hunk 2 above)
Comment (joe-explr) [reply lib/resources/r4/financial_transaction.yaml:307]: Addressed in c339094cbe932d48ad48a1eec08bab902ffe9245

Comment (alexbeckner1) [reply lib/resources/r4/financial_transaction.yaml:307]: Thanks Joseph. Did you see the additional notes changes CPA engineering requested in the comment on https://jira2.cerner.com/browse/RCFHIRAPI-791? :

- When the value is 'true' the payment will be posted across existing financial activity with any outstanding balances using FIFO logic. If there is no existing financial activity, the system will create a new financial activity for current month and post the payment to it. If there are existing financial activity with no outstanding balance, the system will post the payment to the latest month financial activity.
- When the value is 'false' the system will post the payment to the current month financial activity if exists and if not the system creates a new financial activity for current month and post the payment to it.
- I would update "Only supported when the target is Encounter"  to say "Only supported for the Payment Transaction Type with target as Encounter"
- I would also add a note to say "This extension is only supported for recurring encounter and not other types of encounters like inpatient, outpatient, etc."

Comment (joe-explr) [reply lib/resources/r4/financial_transaction.yaml]: Addressed in be8f1c0e06221800273c9522eaebb20ceb7738c1

Comment (alexbeckner1) [reply lib/resources/r4/financial_transaction.yaml]: Added a few corrections in https://github.com/cerner/fhir.cerner.com/commit/be8f1c0e06221800273c9522eaebb20ceb7738c1

Comment (joe-explr) [reply lib/resources/r4/financial_transaction.yaml]: Addressed in a586dda044536a91d678f12655308908c2ab4f88

**Review by alexbeckner1 — APPROVED** [review]

**Review by SukeshKonjeti — APPROVED** [review]

**Pull request merged**